src/test/java/net/geant/nmaas/portal/service/impl/DomainServiceTest.java

```diff
@@ -372,5 +380,28 @@ void shouldSoftRemoveDomain() {
         assertTrue(deletedDomain.get().isDeleted());
         assertTrue(deletedDomain.get().getName().contains("DELETED"));
     }
+    
+    @Test
+    void shouldRemoveDomainFromAllGroupsOnSoftRemoval() {
+        Domain domain = new Domain(1L, "testdom", "testdom");
```

Comment:
Please also test a case then there are multiple different domains in a group if we are removing one of them